metadata/elasticsearch/src/main/java/com/spotify/heroic/metadata/elasticsearch/MetadataBackendKV.java

```diff
@@ -577,19 +586,29 @@ public DeleteSeries end(
         return Pair.of(tk, tv);
     }
 
-    private static void buildContext(final XContentBuilder b, Series series) throws IOException {
+    private static void buildContext(final XContentBuilder b, Series series, boolean indexResourceIdentifiers) throws IOException {
         b.field(KEY, series.getKey());
 
         b.startArray(TAGS);
         for (final Map.Entry<String, String> entry : series.getTags().entrySet()) {
             b.value(entry.getKey() + TAG_DELIMITER + entry.getValue());
         }
+        if(indexResourceIdentifiers) {
+            for (final Map.Entry<String, String> entry : series.getResource().entrySet()) {
+                b.value(entry.getKey() + TAG_DELIMITER + entry.getValue());
+            }
+        }
         b.endArray();
 
         b.startArray(TAG_KEYS);
         for (final Map.Entry<String, String> entry : series.getTags().entrySet()) {
             b.value(entry.getKey());
         }
+        if(indexResourceIdentifiers) {
```

Comment:
This block just adds the `tags`, `TAG_DELIMITER + entry.getValue()` should be removed.